### PR TITLE
 Add EligibleDelegateStore to DPoS module - Closes #7663 

### LINKED ISSUE #7663: Add EligibleDelegateStore to DPoS module
### Description

- Add EligibleDelegateStore as described in [LIP-0057](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#eligible-delegates-substore-1)
- Add `getKey(address: Buffer, newWeight: bigint)` function to the store
- Add `getTop(count: number)` function to the store
- Add `getAll()` function to the store

### Acceptance Criteria

- All functions are unit tested with in memory store

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #7663 

### How was it solved?

Implement as specified

### How was it tested?

Added unit tests

### CODE REVIEW

File: framework/src/modules/dpos_v2/stores/eligible_delegates.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export class EligibleDelegatesStore extends BaseStore<EligibleDelegate> {
+	public schema = eligibleDelegatesStoreSchema;
+
+	public getKey(address: Buffer, delegateWeight: bigint): Buffer {
+		const buffer = Buffer.alloc(8);
+		buffer.writeBigUInt64BE(delegateWeight);
+		return Buffer.concat([buffer, address]);
+	}
+
+	public async getTop(context: ImmutableStoreGetter, count: number) {
```
[the remark this answers is not in the document]
Comment (shuse2) [reply framework/src/modules/dpos_v2/stores/eligible_delegates.ts:42]: or maybe not since it adds unnecessary loop... 🤔

**Review by shuse2 — APPROVED** [review]

File: framework/src/modules/dpos_v2/stores/eligible_delegates.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Incede) [reply framework/src/modules/dpos_v2/stores/eligible_delegates.ts:42]: I think depends on case basis to justify the extra computation (loop & parsing) vs response time so maybe better to leave it to the caller.

[the remark this answers is not in the document]
Comment (Incede) [reply framework/src/modules/dpos_v2/stores/eligible_delegates.ts:42]: but if you think it's better the other way I can also update it

File: framework/test/unit/modules/dpos_v2/stores/eligible_delegates_store.spec.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { StoreGetter } from '../../../../../src/modules/base_store';
+import { EligibleDelegatesStore } from '../../../../../src/modules/dpos_v2/stores/eligible_delegates';
+import { PrefixedStateReadWriter } from '../../../../../src/state_machine/prefixed_state_read_writer';
+import { InMemoryPrefixedStateDB } from '../../../../../src/testing/in_memory_prefixed_state';
+import { createStoreGetter } from '../../../../../src/testing/utils';
+
+describe('UsedHashOnionStore', () => {
```
Comment (has5aan) [inline framework/test/unit/modules/dpos_v2/stores/eligible_delegates_store.spec.ts:21]: Pretty trivial, but probably the description can be modified? 🤔

**Review by has5aan — APPROVED** [review]

**Pull request merged**